File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        from collections import deque
        if root==None:
            return 0
        ans=0
        q=deque([root])
        while q:
            n=len(q)
            
            for i in range(n):
                node=q.popleft()
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            ans+=1  
        return ans              
        

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
class Solution {
    public int maxDepth(TreeNode root) {
        return helper(root);
    }  // ← Missing closing brace here
    
    private int helper(TreeNode node){
        if (node == null){
            return 0;
        }
        int left = helper(node.left);
        int right = helper(node.right);
        return Math.max(left, right) + 1;
    }
}